workers/unity/Packages/io.improbable.gdk.tools/GenerateCode.cs

```
@@ -476,5 +450,94 @@ private static IEnumerable<string> FindDirInPackages(string directory)
+            }
+
+            var codegenOutputPath = Path.GetFullPath(Path.Combine(Application.dataPath, "..", toolsConfig.CodegenOutputDir));
+            codegenArgs.Add($"--native-output-dir=\"{codegenOutputPath}\"");
+
+            // Add user defined schema directories, warn if directory does not exist
+            foreach (var schemaDir in toolsConfig.SchemaSourceDirs)
+            {
+                if (!Directory.Exists(schemaDir))
+                {
+                    Debug.LogWarning($"Schema directory defined in GDK tools configuration does not exist: {schemaDir}");
+                    continue;
+                }
+
+                codegenArgs.Add($"--schema-path=\"{Path.GetFullPath(schemaDir)}\"");
+            }
+
+            // Add package schema directories
+            codegenArgs.AddRange(FindDirInPackages(SchemaPackageDir)
+                .Select(directory => $"--schema-path=\"{directory}\""));
+
+            // Schema Descriptor
+            codegenArgs.Add($"--descriptor-dir=\"{toolsConfig.DescriptorOutputDir}\"");
+
+            codegenArgs.AddRange(
+                toolsConfig.SerializationOverrides.Select(@override => $"--serialization-override=\"{@override}\""));
+
+            var codegenArgsString = string.Join(" ", codegenArgs);
+
+            // For dotnet run / visual studio
+            try
+            {
+                var csprojXml = XDocument.Load(CodegenExe);
+                var projectNode = csprojXml.Element("Project");
+                var propertyGroup = projectNode.Element("PropertyGroup");
+
+                var args = propertyGroup.Element("StartArguments");
+                args?.Remove();
+
+                propertyGroup.Add(XElement.Parse($"<StartArguments>{codegenArgsString}</StartArguments>"));
+                csprojXml.Save(CodegenExe);
+            }
+            catch
+            {
+                throw new Exception("Unable to update csproj with run configuration.");
```

This should be something like:

```
catch (Exception e)
{
	throw new Exception($"Unable to add run configurations to '{CodegenExe}'", e);
}
```

such that we don't swallow the underlying exception and instead wrap it.